src/driver.cpp

```diff
@@ -443,6 +445,40 @@ void driver::scf::calc_properties(const json &json_prop, Molecule &mol) {
         if (plevel == 1) mrcpp::print::time(1, "Quadrupole moment", t_lap);
     }
 
+    if (json_prop.contains("geometric_derivative")) {
+        t_lap.start();
+        mrcpp::print::header(2, "Computing geometric derivative");
+        for (const auto &item : json_prop["geometric_derivative"].items()) {
+            const auto &id = item.key();
+            const auto &prec = item.value()["precision"];
+            const auto &smoothing = item.value()["smooth_prec"];
+            GeometricDerivative &G = mol.getGeometricDerivative(id);
+
+            for (auto k = 0; k < mol.getNNuclei(); ++k) {
+                const auto nuc_k = nuclei[k];
+                auto Z_k = nuc_k.getCharge();
+                auto R_k = nuc_k.getCoord();
+                auto smooth = detail::nuclear_gradient_smoothing(mol.getNNuclei(), smoothing, Z_k);
+                NuclearGradientOperator h(Z_k, R_k, smooth);
+                h.setup(prec);
+                G.getNuclear().row(k) = Eigen::VectorXd::Zero(3);
```

Comment:
Why not do the `getNuclear` only once outside the double loop? Also, the nuclear and electronic components of the gradient are sized correctly and zeroed the moment you create a `GeometricDerivative`, no need to assign zeroes again row-by-row.

Reply:
If zeroing out is really necessary, you could do `G.getNuclear().row(k) = Eigen::RowVector3d::Zero()` should be an infinitesimal more efficient :smile:

Reply:
Actually, the tensor is now initialized as `nan` in order to separate "not computed" from "computed but zero", so zeroing out is necessary